### ISSUE TITLE
[Bug]: Current user display "No files in here", when userid is only number.

### ISSUE THREAD

**Issue by Uptobillion** [nextcloud/server#43462]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When I upgrade nc from 28.0.1 to 28.0.2, shows:
1. Current user display "No files in here", seems like all files are loosen, **I just test, when I create a account with only contains number (like "12345") , the bug shows.**
2. Upload button is grayed.
3. ios or android client can still upload file or see exist files.

### Steps to reproduce

1. upgrade nc from 28.0.1 to 28.0.2 by command line
2. open webui(userid: 12345)

### Expected behavior

Expect that webui can shows all the exist files.

### Installation method

Community Docker image

### Nextcloud Server version

28

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.2

### Web server

Nginx

### Database engine version

MySQL

### Is this bug present after an update or on a fresh install?

Updated from a MINOR version (ex. 22.1 to 22.2)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [X] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "mysql.utf8mb4": true,
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "127.0.0.1",
            "10.5.0.1",
            "test.example.com"
        ],
        "version": "28.0.2.5",
        "theme": "myself",
        "log_rotate_size": 54857600,
        "knowledgebaseenabled": false,
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "dbtype": "mysql",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379,
            "dbindex": 0,
            "timeout": 0.1
        },
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "filelocking.enabled": true,
        "default_language": "zh_CN",
        "default_phone_region": "CN",
        "default_locale": "zh",
        "enabledPreviewProviders": [
            "OC\\Preview\\Imaginary",
            "OC\\Preview\\GIF",
            "OC\\Preview\\JPEG",
            "OC\\Preview\\MarkDown",
            "OC\\Preview\\MP3",
            "OC\\Preview\\PNG",
            "OC\\Preview\\HEIC",
            "OC\\Preview\\Image"
        ],
        "preview_imaginary_url": "***REMOVED SENSITIVE VALUE***",
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "apps_paths": [
            {
                "path": "\/var\/www\/html\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/www\/html\/custom_apps",
                "url": "\/custom_apps",
                "writable": true
            }
        ],
        "mail_smtpmode": "smtp",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtpauth": true,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "trashbin_retention_obligation": "auto, 3",
        "connectivity_check_domains": [
            "www.nextcloud.com"
        ],
        "logtimezone": "Asia\/Shanghai",
        "maintenance": false,
        "updater.release.channel": "stable",
        "upgrade.disable-web": true,
        "memories.exiftool": "\/var\/www\/html\/custom_apps\/memories\/bin-ext\/exiftool-amd64-musl",
        "memories.vod.ffmpeg": "\/usr\/bin\/ffmpeg",
        "memories.vod.ffprobe": "\/usr\/bin\/ffprobe",
        "memories.vod.path": "\/var\/www\/html\/custom_apps\/memories\/bin-ext\/go-vod-amd64",
        "memories.gis_type": 1,
        "ldapProviderFactory": "OCA\\User_LDAP\\LDAPProviderFactory",
        "memories.vod.vaapi": true,
        "memories.vod.connect": "***REMOVED SENSITIVE VALUE***",
        "loglevel": 0,
        "memories.vod.external": true,
        "memories.exiftool_no_local": true,
        "memories.vod.disable": false,
        "mail_sendmailmode": "smtp",
        "mail_smtpsecure": "",
        "mail_smtpauthtype": "LOGIN",
        "auth.bruteforce.protection.enabled": true,
        "maintenance_window_start": 18
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 2.20.0
  - admin_audit: 1.18.0
  - bruteforcesettings: 2.8.0
  - calendar: 4.6.4
  - circles: 28.0.0-dev
  - cloud_federation_api: 1.11.0
  - collectives: 2.9.2
  - comments: 1.18.0
  - contacts: 5.5.1
  - contactsinteraction: 1.9.0
  - dashboard: 7.8.0
  - dav: 1.29.1
  - deck: 1.12.1
  - drawio: 3.0.2
  - end_to_end_encryption: 1.14.1
  - external: 5.3.1
  - federatedfilesharing: 1.18.0
  - federation: 1.18.0
  - files: 2.0.0
  - files_external: 1.20.0
  - files_pdfviewer: 2.9.0
  - files_reminders: 1.1.0
  - files_retention: 1.17.0
  - files_sharing: 1.20.0
  - files_trashbin: 1.18.0
  - files_versions: 1.21.0
  - groupfolders: 16.0.3
  - impersonate: 1.15.0
  - integration_youtube: 0.1.5
  - ldap_write_support: 1.10.0
  - limit_login_to_ip: 4.0.0
  - logreader: 2.13.0
  - lookup_server_connector: 1.16.0
  - memories: 6.2.2
  - notes: 4.9.2
  - notifications: 2.16.0
  - notify_push: 0.6.9
  - oauth2: 1.16.3
  - onlyoffice: 9.0.0
  - password_policy: 1.18.0
  - photos: 2.4.0
  - previewgenerator: 5.4.0
  - privacy: 1.12.0
  - provisioning_api: 1.18.0
  - recommendations: 2.0.0
  - related_resources: 1.3.0
  - serverinfo: 1.18.0
  - settings: 1.10.1
  - spreed: 18.0.3
  - suspicious_login: 6.0.0
  - systemtags: 1.18.0
  - tasks: 0.15.0
  - text: 3.9.1
  - theming: 2.3.0
  - twofactor_backupcodes: 1.17.0
  - updatenotification: 1.18.0
  - user_ldap: 1.19.0
  - user_status: 1.8.1
  - viewer: 2.2.0
  - workflowengine: 2.10.0
Disabled:
  - encryption: 2.16.0
  - externalportal: 1.2.0 (installed 1.2.0)
  - files_lock: 28.0.1 (installed 28.0.1)
  - files_mindmap: 0.0.30 (installed 0.0.30)
  - firstrunwizard: 2.17.0 (installed 2.17.0)
  - ncdownloader: 1.0.20 (installed 1.0.20)
  - nextcloud_announcements: 1.17.0 (installed 1.10.0)
  - sharebymail: 1.18.0 (installed 1.17.0)
  - support: 1.11.0 (installed 1.6.0)
  - survey_client: 1.16.0 (installed 1.8.0)
  - terms_of_service: 2.3.2 (installed 2.3.2)
  - twofactor_totp: 10.0.0-beta.2 (installed 6.4.0)
  - weather_status: 1.8.0 (installed 1.1.0)
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

```shell
{"reqId":"***REMOVED SENSITIVE VALUE***","level":0,"time":"2024-02-08T23:19:38+08:00","remoteAddr":"***REMOVED SENSITIVE VALUE***","user":"***REMOVED SENSITIVE VALUE***","app":"user_ldap","method":"GET","url":"/apps/files/api/v1/stats","message":"instance OC\\User\\Database user backend.","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36 Edg/121.0.0.0","version":"28.0.2.5","data":{"app":"user_ldap"}}
{"reqId":"***REMOVED SENSITIVE VALUE***","level":0,"time":"2024-02-08T23:19:38+08:00","remoteAddr":"***REMOVED SENSITIVE VALUE***","user":"***REMOVED SENSITIVE VALUE***","app":"user_ldap","method":"GET","url":"/apps/files/api/v1/stats","message":"instance OCA\\User_LDAP\\User_Proxy user backend.","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36 Edg/121.0.0.0","version":"28.0.2.5","data":{"app":"user_ldap"}}
{"reqId":"***REMOVED SENSITIVE VALUE***","level":0,"time":"2024-02-08T23:19:38+08:00","remoteAddr":"***REMOVED SENSITIVE VALUE***","user":"***REMOVED SENSITIVE VALUE***","app":"user_ldap","method":"GET","url":"/apps/files/api/v1/stats","message":"instance OC\\Group\\Database group backend.","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36 Edg/121.0.0.0","version":"28.0.2.5","data":{"app":"user_ldap"}}
{"reqId":"***REMOVED SENSITIVE VALUE***","level":0,"time":"2024-02-08T23:19:38+08:00","remoteAddr":"***REMOVED SENSITIVE VALUE***","user":"***REMOVED SENSITIVE VALUE***","app":"user_ldap","method":"GET","url":"/apps/files/api/v1/stats","message":"instance OCA\\User_LDAP\\Group_Proxy group backend.","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36 Edg/121.0.0.0","version":"28.0.2.5","data":{"app":"user_ldap"}}
{"reqId":"***REMOVED SENSITIVE VALUE***","level":0,"time":"2024-02-08T23:19:38+08:00","remoteAddr":"***REMOVED SENSITIVE VALUE***","user":"***REMOVED SENSITIVE VALUE***","app":"user_ldap","method":"GET","url":"/apps/files/api/v1/stats","message":"Calling LDAP function ldap_explode_dn with parameters [\"***REMOVED SENSITIVE VALUE***\",0]","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36 Edg/121.0.0.0","version":"28.0.2.5","data":{"app":"user_ldap"}}
{"reqId":"***REMOVED SENSITIVE VALUE***","level":0,"time":"2024-02-08T23:19:38+08:00","remoteAddr":"***REMOVED SENSITIVE VALUE***","user":"***REMOVED SENSITIVE VALUE***","app":"user_ldap","method":"GET","url":"/apps/files/api/v1/stats","message":"Registered action 1 to plugin OCA\\LdapWriteSupport\\LDAPUserManager","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36 Edg/121.0.0.0","version":"28.0.2.5","data":{"app":"user_ldap"}}
{"reqId":"***REMOVED SENSITIVE VALUE***","level":0,"time":"2024-02-08T23:19:38+08:00","remoteAddr":"***REMOVED SENSITIVE VALUE***","user":"***REMOVED SENSITIVE VALUE***","app":"user_ldap","method":"GET","url":"/apps/files/api/v1/stats","message":"Registered action 16 to plugin OCA\\LdapWriteSupport\\LDAPUserManager","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36 Edg/121.0.0.0","version":"28.0.2.5","data":{"app":"user_ldap"}}
{"reqId":"***REMOVED SENSITIVE VALUE***","level":0,"time":"2024-02-08T23:19:38+08:00","remoteAddr":"***REMOVED SENSITIVE VALUE***","user":"***REMOVED SENSITIVE VALUE***","app":"user_ldap","method":"GET","url":"/apps/files/api/v1/stats","message":"Registered action 1048576 to plugin OCA\\LdapWriteSupport\\LDAPUserManager","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36 Edg/121.0.0.0","version":"28.0.2.5","data":{"app":"user_ldap"}}
{"reqId":"***REMOVED SENSITIVE VALUE***","level":0,"time":"2024-02-08T23:19:38+08:00","remoteAddr":"***REMOVED SENSITIVE VALUE***","user":"***REMOVED SENSITIVE VALUE***","app":"user_ldap","method":"GET","url":"/apps/files/api/v1/stats","message":"Registered action 16777216 to plugin OCA\\LdapWriteSupport\\LDAPUserManager","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36 Edg/121.0.0.0","version":"28.0.2.5","data":{"app":"user_ldap"}}
{"reqId":"***REMOVED SENSITIVE VALUE***","level":0,"time":"2024-02-08T23:19:38+08:00","remoteAddr":"***REMOVED SENSITIVE VALUE***","user":"***REMOVED SENSITIVE VALUE***","app":"user_ldap","method":"GET","url":"/apps/files/api/v1/stats","message":"Registered action deleteUser to plugin OCA\\LdapWriteSupport\\LDAPUserManager","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36 Edg/121.0.0.0","version":"28.0.2.5","data":{"app":"user_ldap"}}
{"reqId":"***REMOVED SENSITIVE VALUE***","level":0,"time":"2024-02-08T23:19:38+08:00","remoteAddr":"***REMOVED SENSITIVE VALUE***","user":"***REMOVED SENSITIVE VALUE***","app":"user_ldap","method":"GET","url":"/apps/files/api/v1/stats","message":"Registered action 1 to plugin OCA\\LdapWriteSupport\\LDAPGroupManager","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36 Edg/121.0.0.0","version":"28.0.2.5","data":{"app":"user_ldap"}}
{"reqId":"***REMOVED SENSITIVE VALUE***","level":0,"time":"2024-02-08T23:19:38+08:00","remoteAddr":"***REMOVED SENSITIVE VALUE***","user":"***REMOVED SENSITIVE VALUE***","app":"user_ldap","method":"GET","url":"/apps/files/api/v1/stats","message":"Registered action 16 to plugin OCA\\LdapWriteSupport\\LDAPGroupManager","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36 Edg/121.0.0.0","version":"28.0.2.5","data":{"app":"user_ldap"}}
{"reqId":"***REMOVED SENSITIVE VALUE***","level":0,"time":"2024-02-08T23:19:38+08:00","remoteAddr":"***REMOVED SENSITIVE VALUE***","user":"***REMOVED SENSITIVE VALUE***","app":"user_ldap","method":"GET","url":"/apps/files/api/v1/stats","message":"Registered action 256 to plugin OCA\\LdapWriteSupport\\LDAPGroupManager","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36 Edg/121.0.0.0","version":"28.0.2.5","data":{"app":"user_ldap"}}
{"reqId":"***REMOVED SENSITIVE VALUE***","level":0,"time":"2024-02-08T23:19:38+08:00","remoteAddr":"***REMOVED SENSITIVE VALUE***","user":"***REMOVED SENSITIVE VALUE***","app":"user_ldap","method":"GET","url":"/apps/files/api/v1/stats","message":"Registered action 4096 to plugin OCA\\LdapWriteSupport\\LDAPGroupManager","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36 Edg/121.0.0.0","version":"28.0.2.5","data":{"app":"user_ldap"}}
```

**Comment by Uptobillion** [comment]
I just test, when I create a account with only contains number (like "12345") , the bug shows.

**Comment by joshtrichards** [comment]
Fixed in #43318
